app/src/highlighter/index.ts

```diff
@@ -266,6 +266,129 @@ const extensionModes: ReadonlyArray<IModeDefinition> = [
       '.jl': 'text/x-julia',
     },
   },
+  {
+    install: () => import('codemirror/mode/stex/stex'),
+    mappings: {
+      '.tex': 'text/x-stex',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/sparql/sparql'),
+    mappings: {
+      '.rq': 'application/sparql-query',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/stylus/stylus'),
+    mappings: {
+      '.styl': 'text/x-styl',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/soy/soy'),
+    mappings: {
+      '.soy': 'text/x-soy',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/smalltalk/smalltalk'),
+    mappings: {
+      '.st': 'text/x-stsrc',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/slim/slim'),
+    mappings: {
+      '.slim': 'application/x-slim',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/sieve/sieve'),
+    mappings: {
+      '.sieve': 'application/sieve',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/scheme/scheme'),
+    mappings: {
+      '.ss': 'text/x-scheme',
+      '.sls': 'text/x-scheme',
+      '.scm': 'text/x-scheme',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/rst/rst'),
+    mappings: {
+      '.rst': 'text/x-rst',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/rpm/rpm'),
+    mappings: {
+      '.rpm': 'text/x-rpm-spec',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/q/q'),
+    mappings: {
+      '.q': 'text/x-q',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/puppet/puppet'),
+    mappings: {
+      '.pp': 'text/x-puppet',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/pug/pug'),
+    mappings: {
+      '.pug': 'text/x-pug',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/protobuf/protobuf'),
+    mappings: {
+      '.proto': 'text/x-protobuf',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/properties/properties'),
+    mappings: {
+      '.properties': 'text/x-properties',
+      '.ini': 'text/x-ini',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/pig/pig'),
+    mappings: {
+      '.pig': 'text/x-pig',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/asciiarmor/asciiarmor'),
+    mappings: {
+      '.pgp': 'application/pgp',
+    },
+  },
+  {
+    install: () => import('codemirror/mode/octave/octave'),
+    mappings: {
+      '.m': 'text/x-octave',
```

Comment:
I wonder what happens here with the dual mapping to `.m`
https://github.com/desktop/desktop/blob/63a7062cbf11145bdbd05e6d699fdf67cb0456ea/app/src/highlighter/index.ts#L122

Reply:
Hmm, I am not sure. It seems `.m` is used by a lot (Objective-C, MATLAB, Mathematica, etc) of different languages. But, Objective-C is probably the most popular so I would think it would be best if we only support that one.

Reply:
I chose to remove support for Octave in favor of Objective-C.